ush/python/pygfs/task/aero_emissions.py

```diff
@@ -75,10 +105,261 @@ def execute(workdir: Union[str, os.PathLike], aprun_cmd: str) -> None:
         -------
         None
         """
+        AerosolEmissions.run(workdir)
+
+    @classmethod
+    @logit(logger)
+    def run(cls, workdir: Union[str, os.PathLike], current_date: str = None, forecast_dates: list = None, Config_dict: Dict = {}) -> None:
+        emistype = Config_dict.emistype
+        ratio = Config_dict.ratio
+        climfiles = sort(glob("{}{}".format(Config_dict.climfile_str, "*.nc")))
+        print(Config_dict.data_out['copy'][0][0])
+
+        if emistype.lower() == "qfed":
+            basefile = glob("qfed2.emis_*.nc4")
+        elif emistype.lower() == "gbbepx":
+            basefile = glob("GBBEPx_all01GRID.emissions_v*.nc")
+        elif emistype.lower() == "hfed":
+            basefile = glob("hfed.emis_*.x576_y361.*nc4")
+
+        dset = AerosolEmissions.make_fire_emission(
+            d=current_date,
+            climos=climfiles,
+            ratio=ratio,
+            scale_climo=True,
+            obsfile=basefile)
+        print(dset)
+
+        AerosolEmissions.write_ncf(dset, Config_dict.data_out['copy'][0][0])
+
+    @staticmethod
+    @logit(logger)
+    def open_qfed(fname) -> xr.Dataset:
+        """
+        Open QFED2 fire emissions data
+
+        Parameters
+        ----------
+        fname : str or list of str
+            Path(s) to the QFED2 fire emissions files
+
+        Returns
+        -------
+        xr.Dataset
+            Dataset containing the fire emissions data
+        """
+
+        vrs = ["BC", "CH4", "CO", "CO2", "NH3", "NOx", "OC", "PM2.5", "SO2"]
+        qfed_vars = ["bc", "ch4", "co", "co2", "nh3", "no", "oc", "pm25", "so2"]
+
+        if len(fname) > 1:
+            files = sort(fname)
+        else:
+            files = sort(glob(fname))
+
+        found_species = []
+        dset_dict = {}
+        for f in files:
+            index_good = [[i, v] for i, v in enumerate(qfed_vars) if v in f]
+            good = index_good[0][0]
+            found_species.append(index_good[0][1])
+            da = xr.open_dataset(f, decode_cf=False).biomass
+            da.name = vrs[good]
+            dset_dict[vrs[good]] = da
+
+        dset = xr.Dataset(dset_dict)
+        return dset
+
+    @staticmethod
+    @logit(logger)
+    def open_climatology(fname) -> xr.Dataset:
+        # array to house datasets
+        das = []
+        # print("")
+        # print("Opening Climatology Files...")
+
+        if len(fname) > 1:
+            files = sort(fname)
+        else:
+            files = sort(glob(fname))
+        # print(files)
+        xr.open_dataset(files[0])
+        for i, f in enumerate(files):
+            # print("  opening:", f)
```

Comment:
A `logger.info` would be nice to have here to indicate while file is being processed.